### DISCUSSION TITLE
Understanding Negation - QA for CondaQA

### DISCUSSION THREAD

**Discussion by eduardoblanco** [eduardoblanco/csc696H-Spring24#8] [General]
See [Understanding Negation](https://github.com/eduardoblanco/csc696H-Spring24/wiki/Understanding-Negation)

**Comment by eduardoblanco** [comment]
(Section 3 from the paper)

> "We find that the affirmative edits change the answers of 77.7% of questions from the original Wikipedia passage

I don't understand why the affirmative edit does not always change the answer. But that could actually be good to avoid spurious artifacts.

If the affirmative edit removes the negation, it changes the meaning (What CondaQA calls "affirmative edit" is not what "Leveraging Affirmative Interpretations" calls "affirmative interpretation"). So doesn't the fact that sometimes the negation goes away and the answer doesn't change mean that the question didn't really need negation understanding?

Who can find an example in CondaQA where the answer to a question in the original passage and the "affirmative edit" is the same?

**Comment by mayanks43** [comment]
> Contrastive groups: closely related passages but with different answers to the same question.
>
> Q: How do you check for "close relatedness"?

I think any two paragraphs talking about the same concepts. For example, the definition of some scientific concept such as "photosynthesis" from two separate textbooks. If we edit one paragraph to reverse an important statement such as plants emit Carbon dioxide instead of Oxygen, than the answer to the question "What gas do the plants emit during photosynthesis?" would change. I am not sure there is a metric that can check for close relatedness. Perhaps, embedding the paragraphs in latent space and checking for embedding similarity could be useful but then, deciding the threshold for the similarity score would be a difficult task. One possible idea is to look at multiple definition websites such as Wikipedia, Britannica, Khan Academy, etc. for definitions of different concepts and try to play with the meaning of the sentences (maybe generate negated statements through rules). 

> "Questions should probe the extent to which models are sensitive to how the negation is expressed."
>
> Q: I don't get the difference with (3): "The dataset should feature contrastive groups: passages that are closely-related, but that may admit different answers to questions, in order to reduce models’ reliance on potential spurious cues in the data and to enable more robust evaluation"
>
> Can you have one without the other?

I think these two points are inter-connected. The first is putting a constraint on the paragraphs and the second on the questions but they are essentially the same point. 

As a whole, I think the paraphrase edit is different from scope and affirmation edits in that the paraphrase edit cannot admit different answers. 

Also, there can be many more kinds of edits that don't involve negation such as the one I suggested above where we make plants emit Carbon Dioxide during photosynthesis.

**Reply by devsawgrad** [reply]
> I am not sure there is a metric that can check for close relatedness. Perhaps, embedding the paragraphs in latent space and checking for >embedding similarity could be useful but then, deciding the threshold for the similarity score would be a difficult task.

Yes, semantic similarity is one option. There's also lexical similarity which given that the paper wanted to test if the model understands implications without relying on specific words/cues would be appropriate. Not sure though if there is a metric that also captures position well.

**Comment by bhavya2109sharma** [comment]
>Crowdworker recruitment. I hadn't heard of Crowdaq before. There is a nice description but no justification. The decision are sound though. Q: Why a small pool?

According to my understanding, although the study does not specifically justify the tiny pool size, it does point to the necessity of specialized knowledge and abilities for the difficult process of editing and questioning around negation. 

>: They pay more to "creative workers". What is a such a worker, and how do they know?

When it comes to "creative workers," these are probably crowdworkers who formulate queries on the implications of negated statements with a greater degree of creativity or thought. The identification of these workers is not explained in depth in the paper, but the fact that they receive more pay implies that their contributions are valued more highly or call for a higher level of expertise.

**Comment by Urvika-gola** [comment]
>Q: How do you check for "close relatedness"?

To ensure close relatedness, only the subject related to the negation is changed by making minimal edits. This ensured that the fundamental context doesn't change too much. And to check for close relatedness. We've used similarity score in vector DB using cosine similarlity to find closely related "words". But we are talking about similar sentences, not similar words in this case, so I am not sure how to do it. CuriousI looked it up and found this SentenceTransformers Python framework [sbert](https://www.sbert.net/) that can be used to calculate embeddings of sentences/ paragraphs to  find how similar two texts are.

> Q: What is a "implication of the negated statement"?

It refers to the resulting meaning or inference that arises when a statement is negated. An example in the paper says "Nearly all of his possessions were destroyed with the exception of a guitar and a prized Jaguar automobile." this **implies** that after this situation, he can still use his guitar and can still drive his car.

> Q: They pay more to "creative workers". What is a such a worker, and how do they know?

They have mentioned that they encourage or reward the construction of good questions about implications by offering multiple _examples_ of what these 'good' questions are, and the verifiers evaluate this probably during verification?
